Code/Framework/AzCore/AzCore/Component/EntitySerializer.cpp

```diff
@@ -207,4 +209,36 @@ namespace AZ
             }
         }
     }
+
+    void DeprecatedComponentMetadata::SetShouldTrackDeprecatedCallback(ShouldTrackDeprecatedCallback callback)
+    {
+        m_shouldTrackDeprecatedCallback = callback;
```

Comment:
nit: The callback name sounds more like a boolean rather than a function. Maybe call this m_trackDeprecatedComponentsCallback ?